evm_loader/program/src/types/transaction.rs

```diff
@@ -96,7 +114,7 @@ impl rlp::Decodable for Transaction {
         let hash = solana_program::keccak::hash(rlp.as_raw()).to_bytes();
```

Comment:
This hash should be calculated from an entire binary blob.
For Legacy transactions, it's `rlp.as_raw()`.
But for Legacy wrapped into Typed Envelope, this should be `hashv(&[&[0x00], rlp.as_raw()])`

Reply:
extracted signature to a separate component